soda/core/soda/execution/data_source.py

```diff
@@ -551,25 +551,50 @@ def sql_information_schema_columns(self) -> str:
     def sql_analyze_table(self, table: str) -> str | None:
         return None
 
-    def sql_get_duplicates(
+    def sql_get_duplicates_count(
         self,
         column_names: str,
         table_name: str,
         filter: str,
-        limit: str | None = None,
-        invert_condition: bool = False,
     ) -> str | None:
-        sql = f"""WITH frequencies AS (
+        sql = dedent(
+            f"""
+            WITH frequencies AS (
               SELECT {column_names}, COUNT(*) AS frequency
               FROM {table_name}
               WHERE {filter}
               GROUP BY {column_names})
             SELECT {column_names}, frequency
             FROM frequencies
-            WHERE frequency {'<=' if invert_condition else '>'} 1"""
+            WHERE frequency > 1
+            ORDER BY frequency DESC"""
+        )
+
+        return sql
+
+    def sql_get_duplicates(
+        self,
+        column_names: str,
+        table_name: str,
+        filter: str,
+        limit: str | None = None,
+        invert_condition: bool = False,
+    ) -> str | None:
+        sql = dedent(
+            f"""
+            WITH frequencies AS (
+                SELECT {column_names}, COUNT(*) AS frequency
+                FROM {table_name}
+                WHERE {filter}
+                GROUP BY {column_names})
+            SELECT {column_names}, frequency
+            FROM frequencies
+            WHERE frequency {'<=' if invert_condition else '>'} 1
+            ORDER BY frequency DESC"""
```

Comment:
@vijaykiran FYI I added order by to both duplicate queries, it seems to make sense especially for the failed rows one with limit to get the ones with most duplicates